base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -2951,7 +2899,25 @@ fn run_migrations(db: &LMDBDatabase) -> Result<(), ChainStorageError> {
             txn.commit()?;
             info!(target: LOG_TARGET, "added RX vm key 91ef83186cefaa646dc4c6e950e68e4debab52b4f4a9b7f465891e91fe5f6ce4");
         }
+        if migrate_from_version == 4 {
```

Comment:
Making `if clear_bad_blocks {` a function within this function may be better, then we can call it in place of toggling the bool, pass in the db write lock as a parameter and keep the commit within the `migrate_from_version` scope.